lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -3632,7 +3649,7 @@ SILValue AdjointEmitter::accumulateMaterializedAdjointsDirect(SILValue lhs,
   assert(lhs->getType() == rhs->getType() && "Adjoints must have equal types!");
   assert(lhs->getType().isObject() && rhs->getType().isObject() &&
          "Adjoint types must be both object types!");
-  auto adjointTy = lhs->getType();
+  auto adjointTy = remapType(lhs->getType());
```

Comment:
Since `lhs` and `rhs` are both expected to be adjoints, why is `remapType` necessary?

Reply:
Thanks! Done in https://github.com/apple/swift/pull/22080/commits/e3c8cb65ccd88c1533e1822f60ae15e60beb0b57.